internal/core/app_exec.go

```diff
@@ -0,0 +1,257 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/hashicorp/go-argmapper"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl/v2"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint/internal/ceb"
+	"github.com/hashicorp/waypoint/internal/config"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+// Exec launches an exec plugin
+func (a *App) Exec(ctx context.Context, id string, d *pb.Deployment) error {
+	// We need to get the pushed artifact if it isn't loaded.
+	// TODO(evanphx) I don't understand why we do this. We're mimicing what the
+	// deploy destroy code does and they load this, so we do to?
```

Comment:
To address this comment: this is for [this feature](https://www.waypointproject.io/docs/waypoint-hcl/variables/artifact) (and similar).

Reply:
Gotcha, thanks!